### ISSUE TITLE
Problem using correlation surface

### ISSUE THREAD

**Issue by alfbr** [CRAVA/crava#230]
We have a case where single surface inversion runs through, but using separate correlation surface fails. Rainer can provide input for reproduction.

**Comment by hgolsen** [comment]
We can start to look at this bug if you provide a data set to NR ftp server.

**Comment by hgolsen** [comment]
Can you provide the data case for this issue @RainerTonn? (I don't think we have received it yet).

**Comment by alfbr** [comment]
Case uploaded in tar.gz by ftp. There were two xml-files, not sure which one we should focus on. Let's do a video meeting once you have had a look.

**Comment by hgolsen** [comment]
The xml-file "BDN_Nov16_LSdir.xml" does not match the data in the input-folder. The "BDN_Nov16.xml" file almost matches the input-files (ex. xml file says "Mid_78_81.sgy" and input folder has file "Mid78_81_2s5s.sgy").

Using "BDN_Nov16.xml" (with updated names) we get a 'Segmentation Fault'-error, but that was due to an error when reading Near78_81_2s5s.sgy:
"_There is a region between the base surface of the inversion interval and the seismic data volume with no seismic data. The largest gap is for the seismic trace at position (842048,5316951) where the surface z-value = 3888.09 while the seismic end time = 3004.00. Please include more seismic data or heighten the base surface 884.09ms._"

Does this sounds familiar? Can you take a look at the segy-data?

We will update Crava so the user gets this message and not the 'segmentation fault'-error. However, this isn't the bug in question..

**Comment by RainerTonn** [comment]
Seismic and horizon grid are of different size. I reduced the working area. First to the size of the seismic grit and then even smaller. The horizon onlap error occurred even with the small size. The big area has problems with the horizon. Due to extrapolation there might be zones where the horizons are crossing.

**Comment by RainerTonn** [comment]
here are the dimensions
![cr1](https://cloud.githubusercontent.com/assets/19913195/16449416/fda2b76a-3dd1-11e6-96b4-1463dde87378.PNG)

**Comment by RainerTonn** [comment]
Sorry for the confusion with the xml files. I created so many, probably I lost track. I believe this is now the correct one (renamed to *xml.txt
[outputD4_859TESTxml.txt](https://github.com/CRAVA/crava/files/339072/outputD4_859TESTxml.txt)
)

:
[D4_859TSETlogFile.txt](https://github.com/CRAVA/crava/files/339065/D4_859TSETlogFile.txt)

**Comment by hgolsen** [comment]
I have tried to run a few versions of this case:

Case1:
- If I run with \<interval-one-surface> and with thickness 500, there was a small bug (with and without \<correlation-direction>). We have fixed this and this now runs ok.

Case2:
- When I ran with \<interval-two-surfaces> with \<correlation-direction> (like the attached xml) I found a bug with estimation of prior-correlations. We have fixed this (but not checked in), and updated with an error message: "There is not enough well data for estimation of prior correlations. For estimation of prior correlations at least 100 blocks are needed; currently there are 58. Either increase the number of layers or add wells."

I have tried to increase the number of layers (to 30), and it worked.

Case3:
- If I run \<interval-two-surfaces> without \<correlation-direction> I get an error message:
-Error with top/bottom grids. Minimum thickness should be at least 0.41 times maximum, it is 0.338951.

This is due to the issue with the surfaces (as you mentioned). One solution is to add correlation-direction (which leads to Case2).
Decreasing area size (e.g. setting \<xl-end> to 6300), instead of adding correlation-direction, also works (but this will also lead to the prior-correlation issue in Case2).
